dev/test/intersection.test.ts

```diff
@@ -77,6 +83,32 @@ describe("intersection", () => {
                 object: { props: { a: "string", b: "boolean" } }
             })
         })
+        describe("Tuple and Array", () => {
+            it("string type", () => {
```

Comment:
Unfortunately this doesn't really test that the intersection works because the tuple is not changed. The intersected array should add something more specific, e.g. `alpha[]`. At that point, you don't actually need to check traversal (unless it's required for coverage), you can just snap the output node and make sure it is a tuple of two `alpha` strings.